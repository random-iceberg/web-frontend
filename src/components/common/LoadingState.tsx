/*
Draft Doc

This Component is for Loading States, Use this while content is loading.
*/
import React from "react";

interface LoadingStateProps {
  message: string;
}

const LoadingState: React.FC<LoadingStateProps> = ({ message }) => {
  return <div className="p-8 text-center bg-gray-50 rounded-lg">{message}</div>;
};

export default LoadingState;
