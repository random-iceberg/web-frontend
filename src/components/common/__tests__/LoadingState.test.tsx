import React from "react";
import { render } from "@testing-library/react";
import LoadingState from "../LoadingState";

describe("LoadingState Component", () => {
  it("renders correctly with a message", () => {
    const { asFragment } = render(<LoadingState message="Loading data..." />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with a different message", () => {
    const { asFragment } = render(<LoadingState message="Fetching results..." />);
    expect(asFragment()).toMatchSnapshot();
  });
});
