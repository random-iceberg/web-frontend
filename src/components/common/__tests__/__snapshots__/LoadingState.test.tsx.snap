// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingState Component renders correctly with a different message 1`] = `
<DocumentFragment>
  <div
    class="p-8 text-center bg-gray-50 rounded-lg"
  >
    Fetching results...
  </div>
</DocumentFragment>
`;

exports[`LoadingState Component renders correctly with a message 1`] = `
<DocumentFragment>
  <div
    class="p-8 text-center bg-gray-50 rounded-lg"
  >
    Loading data...
  </div>
</DocumentFragment>
`;
